### ISSUE TITLE
issue with keyval completion and open bracket

### ISSUE THREAD

**Issue by mbertucci47** [texstudio-org/texstudio#3159]
### Environment
- TeXstudio: 4.6.0alpha1
- Qt: 6.5.0
- OS: Windows 10
- TeX distribution: N/A

### Expected behavior
Once one selects a keyval with "Enter" or by clicking it, the completer should not pop back up with the keyval list.

### Actual behavior
If the argument is not closed, the keyval list pops up. Consider

<img width="415" alt="pic1" src="https://github.com/texstudio-org/texstudio/assets/61854785/e19c8a68-4fc8-4034-b73b-acb71f6a4939">

Once you select the keyval, the completion list should disappear because the user needs to fill the "val" of the "keyval". However after selecting the keyval it just pops back up:

<img width="478" alt="pic2" src="https://github.com/texstudio-org/texstudio/assets/61854785/8d08e077-4f06-4c35-a64f-307f88ef84e8">

The issue does not occur if the argument is closed, meaning starting with `\begin{tikzcd}[fon]`. The same behavior is observed regardless of the delimiter, e.g. `\tikzcdset{fon` shows the same issue.

This is relevant for users like me who don't have the completer turned on by default (so closing delimiters are not added automatically) but do often activate the keyval completion manually with `Ctrl+Space`.

### How to reproduce

```latex
\documentclass{article}
\usepackage{tikz-cd}

\begin{document}

\begin{tikzcd}[fon

\end{tikzcd}
\end{document}
```
Insert cursor after `fon` and follow steps above.

**Comment by mbertucci47** [comment]
It seems the option you're talking about is "Insert arguments" which does fix it - there is also an "Arguments as placeholders" option; it seems to have no effect on this and I don't know what it does in general.

I'm not a fan of the placeholders but no worries if you don't think this is worth fixing for users with placeholders disabled.

**Comment by mbertucci47** [comment]
Actually from the [documentation](https://texstudio-org.github.io/configuration.html#configuring-the-autocompletion) it would seem "Arguments as placeholders" should be the option which controls the placeholder feature. But I just double checked and it's "Insert arguments" that determines whether or not a placeholder is created. Is this intended behavior?

**Comment by sunderme** [comment]
Probably you need some text for the placeholders for them to work.

**Comment by sunderme** [comment]
I have fixed some problems around this issue, but the actual problem is the "unlosed" square bracket.
Usually when using the completer, brackets always appear in pairs and there is no issue.

**Comment by mbertucci47** [comment]
Thanks! Right, I admit I'm a somewhat odd user in wanting keyval completion without using command completion much. I'll test the new build

**Comment by mbertucci47** [comment]
As far as I can tell the behavior is the same as before, though perhaps that's what you meant by
> the actual problem is the "unlosed" square bracket

**Comment by sunderme** [comment]
yes, I can't fix the issue of the "unclosed" bracket easily, hence the remark.

**Comment by sunderme** [comment]
Seems to be fixed

**Comment by mbertucci47** [comment]
@sunderme Do you mean in the current release or in unreleased code? Because I still see the issue in 4.9.0beta2

**Comment by sunderme** [comment]
current head in git

**Comment by mbertucci47** [comment]
The issue is still present with 4.9.0beta3:

<img width="547" height="401" alt="Image" src="https://github.com/user-attachments/assets/813ff9a1-997d-4b05-aa1d-32b5180ece17" />

Also, there's a new issue. Colors from beamer are populating every keyval list. That's where `alert` and `averagebackgroundcolor` are coming from in the image above. I will report this separately. But this issue should be reopened I think.

**Comment by sunderme** [comment]
I don't see the completion 'alert',etc.
(4.9.0beta3,win)

**Comment by sunderme** [comment]
seems to fixed now

**Comment by mbertucci47** [comment]
Confirmed. Thanks!
